test.cpp

```diff
@@ -3231,6 +3231,15 @@ static void fuzz_crash()
     }
 }
 
+static void leak()
+{
+    {
+        const char code[] = "#define e(...)__VA_OPT__()\n"
+                            "#define e\n";
+        (void)preprocess(code, simplecpp::DUI());
```

Comment:
I assume this has to be executed by some valgrind or sanitizer to detect the issue? i.e. if I just compile and run then this test does not ensure there is no leak?
It feels a bit misplaced somehow but maybe writing a separate test for it is overkill.. but well a comment would be nice?

Reply:
I added some comments.